libzmix/benches/bbs_vs_ps.rs

```diff
@@ -29,16 +37,62 @@ fn sign_messages_benchmark(c: &mut Criterion) {
             b.iter(|| BBSSignature::new(attributes.as_slice(), &sk, &pk))
         });
         let label = format!("ps sign {} atts", atts);
-        let (pk, sk) = ps_keys_generate(atts, label.as_bytes());
+        let params = Params::new(label.as_bytes());
+        let (_, sk) = ps_keys_generate(atts, &params);
         c.bench_function(label.as_str(), |b| {
-            b.iter(|| PSSignature::new(attributes.as_slice(), &sk, &pk))
+            b.iter(|| PSSignature::new(attributes.as_slice(), &sk, &params))
         });
     }
 }
 
-fn bbs_sign_committed_messages_benchmark(c: &mut Criterion) {}
+fn bbs_sign_committed_messages_benchmark(c: &mut Criterion) {
+
+}
+
+fn bbs_prove_benchmark(c: &mut Criterion) {
+    for atts in vec![1, 2, 5, 10, 20, 50, 100, 200] {
+        ////////////////////////// BBS+ Signatures
+        let (pk, sk) = bbs_keys_generate(atts).unwrap();
+        let attributes = SignatureMessageVector::random(atts);
+        let sig = BBSSignature::new(attributes.as_slice(), &sk, &pk).unwrap();
+
+        c.bench_function(format!("bbs+ generate proof {} atts", atts).as_str(), |b| {
+            b.iter(|| {
+                let pok = BBSPoKOfSignature::init(&sig, &pk, attributes.as_slice(), None, HashSet::new()).unwrap();
+                let challenge = FieldElement::from_msg_hash(&pok.to_bytes());
+                pok.gen_proof(&challenge).unwrap()
+            })
+        });
+        let poks = BBSPoKOfSignature::init(&sig, &pk, attributes.as_slice(), None, HashSet::new()).unwrap();
+        let challenge = FieldElement::from_msg_hash(&poks.to_bytes());
+        let proof = poks.gen_proof(&challenge).unwrap();
+        c.bench_function(format!("bbs+ verify proof {} atts", atts).as_str(), |b| {
+            b.iter(|| proof.verify(&pk, HashMap::new(), &challenge).unwrap())
+        });
+
+        ////////////////////////// PS Signatures
+        let label = format!("ps verify proof {} atts", atts);
+        let params = Params::new(label.as_bytes());
+        let (pk, sk) = ps_keys_generate(atts, &params);
+        let sig = PSSignature::new(attributes.as_slice(), &sk, &params).unwrap();
+
+        c.bench_function(format!("ps generate proof {} atts", atts).as_str(), |b| {
+            b.iter(|| {
+                let pok = PSPoKOfSignature::init(&sig, &pk, &params, attributes.as_slice(), None, HashSet::new()).unwrap();
+                let chal = FieldElement::from_msg_hash(&pok.to_bytes());
+                pok.gen_proof(&chal).unwrap()
+            })
+        });
+
+        let pok = PSPoKOfSignature::init(&sig, &pk, &params, attributes.as_slice(), None, HashSet::new()).unwrap();
+        let chal = FieldElement::from_msg_hash(&pok.to_bytes());
+        let proof = pok.gen_proof(&chal).unwrap();
 
-fn bbs_prove_benchmark(c: &mut Criterion) {}
+        c.bench_function(label.as_str(), |b| {
```

Comment:
Why do you use `label` as the id and not something like `format!("ps verify proof {} atts", atts)` the way you do for BBS+ sigs above?

Reply:
its used as the label to the params. Doesn't make much of a difference